src/include/migraphx/algorithm.hpp

```diff
@@ -76,6 +96,53 @@ void group_unique(Iterator start, Iterator last, Output out, Predicate pred)
     }
 }
 
+template <class Iterator, class Predicate, class Output>
+void group_find(Iterator start, Iterator last, Predicate pred, Output out)
+{
+    start = std::find_if(start, last, pred);
+    while(start != last)
+    {
+        auto it = std::find_if_not(start, last, pred);
+        out(start, it);
+        start = std::find_if(it, last, pred);
+    }
+}
+
+template <class Iterator, class Predicate>
+Iterator adjacent_remove_if(Iterator first, Iterator last, Predicate p)
```

Comment:
Can you add docstrings for all the new additions in `algorithm.hpp` ? 
e.g. this one has poitner arithmetics making it difficult to read.

Reply:
> e.g. this one has pointer arithmetics making it difficult to read code.

It was also difficult to write which is why I added unit tests for it.